app/schemas/interactions/nudged.avsc

```diff
@@ -0,0 +1,34 @@
+{
+  "name": "interactions.nudged",
+  "type": "record",
+  "fields": [
+    {
+      "name": "uuid",
+      "type": "string"
```

Comment:
Can we store all UUIDs as `{"type": "fixed", "size": 16}` for compactness?  A String form takes 36 characters (32 hex characters plus four hyphens) for a total of 288 bits whereas the binary data is just 16 octets for a total of 128 bits.

Reply:
At first I was "obviously compact is best", but thinking harder about it, I realize we're talking about the data-as-sent-to-API server. No, I'm _not_ a fan of specifying maximally-compact binary formats for client generation. It's not the obvious thing to do, for an HTTPS transported data stream. For data-at-rest in the kafka-store, sure, let's compact it. I'm not even sure off the top of my head _how_ to encode a binary blob to send to the REST endpoint. So, I'm starting to think we may want to be defining two encodings and a transform: data-in-flight, and data-at-rest

Reply:
@reedstrm I think what @jpslav is referring to here is just the storage of a fixed type of size 16.  the encoding would convert the passed in data (in string form) into this byte form (I believe).   decoding does the opposite.

Reply:
@jpslav i have not gotten the ^ to work yet.  i get errors when i pass variations of `{"type": "fixed", "size": 16}` to fields...i created an issue so i can get an answer hopefully from the gem author.

Reply:
for compact storage I’m only talking about encoding for storage in Kafka.  Data as sent to API server should be UUID strings and ISO8601 (or similar) timestamp strings.

Reply:
@jpslav got the uuid storing as 16 bytes now